### PR TITLE
`bzip2.rs`: use the standard library for input/output streams (when they are not passed to bzip itself)

### PR DESCRIPTION
**Pull request by folkertdev**

### CODE REVIEW

File: bzip2.rs [hunk 1]
```diff
@@ -973,14 +955,29 @@ fn apply_saved_time_info_to_output_file(_dst_name: &CStr, _metadata: Metadata) {
...
         if fd < 0 {
             // diverges
             ioError()
         }
 
+        set_permissions_fd(fd, metadata)
+    }
+}
+
+unsafe fn set_permissions_rust(file: &std::fs::File, metadata: &Metadata) {
```
Comment (bjorn3) [inline bzip2.rs:960]: _file and _metadata to avoid warnings on Windows.

File: bzip2.rs [hunk 2]
```diff
@@ -973,14 +955,29 @@ fn apply_saved_time_info_to_output_file(_dst_name: &CStr, _metadata: Metadata) {
...
+    #[cfg(unix)]
+    {
+        use std::os::fd::AsRawFd;
+        set_permissions_fd(file.as_raw_fd(), metadata)
+    }
+}
+
+unsafe fn set_permissions_fd(fd: core::ffi::c_int, metadata: &Metadata) {
```
Comment (bjorn3) [inline bzip2.rs:968]: _fd and _metadata

File: bzip2.rs [hunk 3]
```diff
@@ -1251,9 +1248,37 @@ unsafe fn compress(name: Option<&str>) {
...
 
+enum OutputStream {
+    Stdout(std::io::Stdout),
+    File(std::fs::File),
+}
+
+impl std::io::Write for OutputStream {
+    fn write(&mut self, buf: &[u8]) -> std::io::Result<usize> {
+        match self {
+            OutputStream::Stdout(_stdout) => _stdout.write(buf),
```
Comment (bjorn3) [inline bzip2.rs:1243]: s/_stdout/stdout here and below

File: bzip2.rs [hunk 4]
```diff
@@ -120,7 +120,21 @@ unsafe fn myfeof(f: *mut FILE) -> Bool {
...
-unsafe fn compressStream(stream: *mut FILE, zStream: *mut FILE, metadata: Option<&Metadata>) {
+enum InputStream {
+    Stdin(std::io::Stdin),
+    File(std::fs::File),
+}
+
+impl std::io::Read for InputStream {
+    fn read(&mut self, buf: &mut [u8]) -> std::io::Result<usize> {
+        match self {
+            InputStream::Stdin(_stdin) => _stdin.read(buf),
```
Comment (bjorn3) [inline bzip2.rs:131]: s/_stdin/stdin

File: bzip2.rs (hunk 3 above)
Comment (folkertdev) [reply bzip2.rs:1243]: stdin and stdout are statics in this commit, and cannot be shadowed. so these can only be renamed after your windows PR

Comment (folkertdev) [reply bzip2.rs:1310]: actually, we still (on unix) have a static named `stdin`. We could work around that with `link_name`?

**Review by bjorn3 — APPROVED** [review]

**Pull request merged**